pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarClientImpl.java

```diff
@@ -718,13 +719,11 @@ public void shutdown() throws PulsarClientException {
                 log.warn("Failed to shutdown eventLoopGroup", t);
                 throwable = t;
             }
-            if (createdCnxPool) {
-                try {
-                    cnxPool.close();
-                } catch (Throwable t) {
-                    log.warn("Failed to shutdown cnxPool", t);
-                    throwable = t;
-                }
+            try {
+                closeCnxPool(cnxPool);
+            } catch (Throwable t) {
```

Comment:
what about catching PulsarClientException and not a raw Throwable ?

Reply:
okay. should I change it everywhere in this function? @eolivelli

Reply:
I won't change other points, the effect may be unpredictable.

in your new function `closeCnxPool` you are already catching Throwable so here it is useless.

catching `Throwable` and `Exception` is always a code smell and we should do it only in very specific cases when we know what we are doing.

Reply:
alright. thanks. I have changed it

Reply:
@eolivelli I changed at 2 other places as well in this function and I believe it is safe as I had done those changes a month ago and we haven't released after that,